ReCaptchaUi/Model/UiConfigResolver.php

```diff
@@ -30,22 +36,18 @@ class UiConfigResolver implements UiConfigResolverInterface
     /**
      * @param CaptchaTypeResolverInterface $captchaTypeResolver
      * @param UiConfigProviderInterface[] $uiConfigProviders
-     * @throws InputException
      */
     public function __construct(
         CaptchaTypeResolverInterface $captchaTypeResolver,
         array $uiConfigProviders = []
     ) {
         $this->captchaTypeResolver = $captchaTypeResolver;
-
-        foreach ($uiConfigProviders as $uiConfigProvider) {
-            if (!$uiConfigProvider instanceof UiConfigProviderInterface) {
-                throw new InputException(
-                    __('UI config provider must implement %1', [ UiConfigResolverInterface::class])
-                );
-            }
-        }
-        $this->uiConfigProviders = $uiConfigProviders;
+        $this->uiConfigProviders = array_combine(
```

Comment:
I do like the functional approach but it's much less performance and much less clear what is going on here. I would switch this back.

Reply:
we could get rid of this totally, as the get method is signed, we are sure that the correct type should be returned. Developers must follow the contrat of the doc anyway

Reply:
I don't think we even need to have such validation. There is an implicit contract. I agree we shouldn't add noise when it has no value.

Reply:
@nathanjosiah